rustworkx-core/src/generators/random_graph.rs

```diff
@@ -1402,6 +1402,27 @@ mod tests {
         };
     }
 
+    #[test]
+    fn test_barabasi_albert_graph_seeding() {
+        use petgraph::visit::EdgeRef;
+        let g1: petgraph::graph::UnGraph<(), ()> =
+            barabasi_albert_graph(7, 2, Some(42), None, || (), || ()).unwrap();
+        let g2: petgraph::graph::UnGraph<(), ()> =
+            barabasi_albert_graph(7, 2, Some(42), None, || (), || ()).unwrap();
+        // Same seeds should yield the same graph
+        let edges1: std::collections::BTreeSet<_> = g1
+            .edge_references()
+            .map(|e| (e.source(), e.target()))
+            .collect();
+        let edges2: std::collections::BTreeSet<_> = g2
+            .edge_references()
+            .map(|e| (e.source(), e.target()))
+            .collect();
```

Comment:
If this test is supposed to check deterministic output with a fixed seed I would expect using a vec here because the btreeset will be iterating based on a sorted order:

```suggestion
        let edges1: Vec<_> = g1
            .edge_references()
            .map(|e| (e.source(), e.target()))
            .collect();
        let edges2: Vec<_> = g2
            .edge_references()
            .map(|e| (e.source(), e.target()))
            .collect();
```

Or is the exact creation order not expected to be deterministic and just the edges created?

Reply:
I think it was just that the created edges were deterministic. In the GH issue, we got a report that we are yielding different edges.